src/pybind/mgr/cephadm/module.py

```diff
@@ -2297,17 +2128,17 @@ def _create_daemon(self, daemon_type, daemon_id, host,
         deps = []  # type: List[str]
         cephadm_config = {} # type: Dict[str, Any]
         if daemon_type == 'prometheus':
-            cephadm_config, deps = self._generate_prometheus_config()
+            cephadm_config, deps = self.prometheus_service.generate_config()
             extra_args.extend(['--config-json', '-'])
         elif daemon_type == 'grafana':
-            cephadm_config, deps = self._generate_grafana_config()
+            cephadm_config, deps = self.grafana_service.generate_config()
             extra_args.extend(['--config-json', '-'])
         elif daemon_type == 'nfs':
             cephadm_config, deps = \
-                    self._generate_nfs_config(daemon_type, daemon_id, host)
+                    self.nfs_service._generate_nfs_config(daemon_type, daemon_id, host)
```

Comment:
```suggestion
                    NFSService(self)._generate_nfs_config(daemon_type, daemon_id, host)
```
initialize the service class when used?

Reply:
I'd like to leave it as it is and only have minimal changes to the code. otherwise I'll have to rebase this PR to death